### PR TITLE
SP Idempotency

### LINKED ISSUE #295: server profile update in-place
Having a strange issue-

some background
Creating a server profile based on a template.  The template has bios options in it.
The server profile has local storage and connection settings in it.

When I create TF apply the server profile, it creates exactly as expected. Excellent!
However:
If I rerun TF plan/apply, TF tries to improperly "change" the resource, specifically the bios settings are planned to -> null and  connection settings ->null and then connections wants to do a delete/add operation.

I am wondering if the state file is problematic here on this?
Let me know if you need sample code.

@AsisBagga Thank you so much for all your help getting this going.

### PR DESCRIPTION
**Pull request by AsisBagga**
### Description
In this PR we shall complete the idempotency for below attributes. 

- [x] Connection Settings
- [x] Management Processor
- [x] Bios Option
- [x] Boot Mode
- [x] Boot
- [x] Firmware
- [x] Local Storage

### Issues Resolved
#295 
#370 

NA

### Check List
- [ ] New functionality includes testing.
  - [ ]  All tests pass for go 1.11 + gofmt checks.
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in the examples (please include helpful comments).
  - [ ] New endpoints supported are updated in the endpoints-support.md file.
- [ ] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: oneview/resource_server_profile.go [hunk 1]
```diff
@@ -92,17 +93,12 @@ func resourceServerProfile() *schema.Resource {
 							Type:     schema.TypeBool,
 							Required: true,
 						},
-						"consistency_state": {
-							Type:     schema.TypeString,
```
Comment (AvijitOS) [inline oneview/resource_server_profile.go:96]: why this is removed?

File: oneview/resource_server_profile.go [hunk 2]
```diff
@@ -523,7 +508,7 @@ func resourceServerProfile() *schema.Resource {
 												},
 												"drive_number": {
 													Type:     schema.TypeInt,
-													Optional: true,
+													Computed: true,
 												},
 												"drive_technology": {
 													Type:     schema.TypeString,
```
Comment (AvijitOS) [inline oneview/resource_server_profile.go:514]: if the local storage is configured from SPT and apply to SP, have we validated? i mean are we not getting any idempotency issues

File: oneview/resource_server_profile.go (hunk 1 above)
Comment (AsisBagga) [reply oneview/resource_server_profile.go:96]: not supported to SP.

Comment (AsisBagga) [reply oneview/resource_server_profile.go:96]: its added back.

**Review by AvijitOS — APPROVED** [review]

**Review by nabhajit-ray — APPROVED** [review]

**Pull request merged**